src/modrover/model.py

```diff
@@ -0,0 +1,271 @@
+from dataclasses import dataclass
+from operator import attrgetter
+from warnings import warn
+
+import numpy as np
+from regmod.data import Data
+from regmod.models import Model as RegmodModel
+from regmod.variable import Variable
+from pandas import DataFrame
+from typing import Dict, List, Optional, Tuple
+
+from .globals import metric_dict
+from .modelid import ModelID
+from .info import model_type_dict
+
+@dataclass
+class Rover:
+
+    # Placeholder, to be fleshed out later and probably moved.
+    model_type: str
+    col_fixed: Dict[str, List]
+    # col_fixed keyed by parameter, ex.
+    # {
+    #     'mu': ['intercept'],
+    #     'sigma': []
+    # }
+    col_covs: List[str]  # All possible covariates we can explore over
+    col_obs: str
+
+
+class Model:
+
+    def __init__(
+        self,
+        model_id: ModelID,
+        model_type: str,
+        col_obs: str,
+        col_covs: List[str],
+        col_fixed: Dict[str, List],
+        col_offset: str = "offset",
```

Comment:
`col_offset` might also need to be a dictionary, since each parameter can have its own offset.